src/liballoc/heap.rs

```diff
@@ -17,9 +17,13 @@
 
 use core::intrinsics::{min_align_of_val, size_of_val};
 use core::mem::{self, ManuallyDrop};
+use core::ptr::NonNull;
 use core::usize;
 
-pub use core::heap::*;
+use core::heap;
+pub use core::heap::{AllocErr, CannotReallocInPlace, CollectionAllocErr, Layout};
+#[cfg(not(stage0))]
+pub use core::heap::{Alloc, Excess};
 #[doc(hidden)]
 pub mod __core {
```

Comment:
Can `#[cfg(stage0)]` be added to this `__core` module?

Reply:
Oh right, because the only use of that `__core` was the `#[global_allocator]` expansion. Neat!